Extensions/GoogleOptimize/Src/Tasks/GoogleOptimize/operations/optimizeclient.ts

```diff
@@ -0,0 +1,61 @@
+import * as hm from 'typed-rest-client/Handlers';
+import * as restm from 'typed-rest-client/RestClient';
+import * as tl from 'azure-pipelines-task-lib/task';
+import {TaskParameter} from './../models/TaskParameter';
+import {IVariation,IExperiment} from './../models/IOptimize';
+import {GetExperimentUtil, UpdateExperimentUtil, Authorize} from './../models/GoogleOptimizeUtils'
+
+let userAgent: string;
+let url: string;
+
+export class Optimizeclient {
+	private param: TaskParameter;
+	private endpointId: string;
+
+	constructor() {
+		this.param = TaskParameter.getInstance();
+		this.endpointId = this.param.endpoint;
+		url = tl.getEndpointUrl(this.endpointId, true);
+		userAgent = "AzDevOps_GoogleOptimize";
+	}
+
+	public async updateExperiment < T extends IExperiment > (experiment: T) {
+		Authorize(async function (err: any, token: any) {
+
+			if(token == null) {
+				console.log(tl.loc("AccessTokenGenerationFailed" , err));
+				tl.setResult(tl.TaskResult.Failed, err);
+			}
+
+			const bearerHandler = new hm.BearerCredentialHandler(token);
+			var restclient = new restm.RestClient(userAgent, url, [bearerHandler]);
+
+			let currentExperiment = await GetExperimentUtil(restclient);
+
+			if (currentExperiment.statusCode != 200) {
+				console.log(tl.loc("FailedToFetchCurrentExperiment", JSON.stringify(currentExperiment.error)));
+				tl.setResult(tl.TaskResult.Failed, err);
+				throw tl.loc("FailedToFetchCurrentExperiment" ,currentExperiment.error );
+			}
+
+			if(currentExperiment.result.status === "ENDED"){
```

Comment:
What if customer wanted to end the experiment but it was already stopped? This will fail the task.

Reply:
Replace ENDED with enum val defined in taskoperation file